src/js/protocols/webstm32.js

```diff
@@ -0,0 +1,941 @@
+                        }
+
+                        const selectedBoard = TABS.firmware_flasher.selectedBoard !== '0' ? TABS.firmware_flasher.selectedBoard : 'NONE';
+                        const connectedBoard = FC.CONFIG.boardName ? FC.CONFIG.boardName : 'UNKNOWN';
+
+                        const reboot = () => {
+                            const buffer = [];
+                            buffer.push8(rebootMode);
+                            setTimeout(() => {
+                                MSP.send_message(MSPCodes.MSP_SET_REBOOT, buffer, () => {
+                                    // if firmware doesn't flush MSP/serial send buffers and gracefully shutdown VCP connections we won't get a reply, so don't wait for it.
+                                    this.msp_connector.disconnect(disconnectionResult => {
+                                        console.log('Disconnecting from MSP', disconnectionResult);
+                                        onDisconnect(disconnectionResult);
+                                    });
+                                });
+                                console.log('Reboot request received by device');
+
+                                // Using onDisconnect to handle the case where the device doesn't respond to the reboot request fixes the callback not being called
+                                onDisconnect(true);
+                            }, 100);
+                        };
+
+                        function onAbort() {
+                            GUI.connect_lock = false;
+                            rebootMode = 0;
+                            console.log('User cancelled because selected target does not match verified board');
+                            reboot();
+                            TABS.firmware_flasher.refresh();
+                        }
+                        try {
+                            if (selectedBoard !== connectedBoard && !TABS.firmware_flasher.localFirmwareLoaded) {
+                                TABS.firmware_flasher.showDialogVerifyBoard(selectedBoard, connectedBoard, onAbort, reboot);
+                            } else {
+                                reboot();
+                            }
+                        } catch (e) {
+                            console.error(e);
+                            reboot();
+                        }
+                    });
+                }
+            }
+
+            const onTimeoutHandler = function () {
+                GUI.connect_lock = false;
+                console.log('Looking for capabilities via MSP failed');
+
+                TABS.firmware_flasher.flashingMessage(i18n.getMessage('stm32RebootingToBootloaderFailed'), TABS.firmware_flasher.FLASH_MESSAGE_TYPES.INVALID);
+            };
+
+            const onFailureHandler = function () {
+                GUI.connect_lock = false;
+                TABS.firmware_flasher.refresh();
+            };
```

Comment:
Similar to the above, can be just a helper function in the same file instead of being inline.

Reply:
Thanks for review.

Done.

Due to second refactor found an issue with MSP still listening - now fixed.